engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -1805,6 +1938,12 @@ static int GetBuffer(lua_State* L)
     void* resource = CheckResource(L, g_ResourceModule.m_Factory, path_hash, "bufferc");
 
     dmGameSystem::BufferResource* buffer_resource = (dmGameSystem::BufferResource*)resource;
+
+    if (!dmBuffer::IsBufferValid(buffer_resource->m_Buffer))
+    {
+        return luaL_error(L, "The buffer handle is invalid");
+    }
```

Comment:
if a reference to a buffer has been removed (e.g a LuaHBuffer object has GC'd), we should tell the user that